Initial OSGi docs

neykov opened the pull request.

sjcorbett commented on guide/misc/osgi.md:
```diff
@@ -0,0 +1,63 @@
+layout: website-normal
+---
+
+# Running Apache Brooklin inside Karaf container
```
`Brooklin`

sjcorbett commented on guide/misc/osgi.md:
```diff
@@ -0,0 +1,63 @@
+
+## Caveats
+
+In the OSGi world specifying class names by string in Brooklyn's configuraiton will work only
```
`configuraiton`

sjcorbett commented:
Looks good. Do you want me to double check any of the instructions in particular?

geomacy commented on guide/misc/osgi.md:
```diff
@@ -0,0 +1,63 @@
+---
+title: FAQ
```
The title is misleading, it's not a FAQ page really - change to 'OSGI'?

m4rkmckenna commented on guide/misc/osgi.md:
```diff
@@ -0,0 +1,63 @@
+{% highlight bash %}
+bin/karaf debug
+{% endhighlight %}
+
```
Might be worth mentioning `karaf clean` to reset config admin / installed features to default

geomacy commented on guide/misc/osgi.md:
```diff
@@ -0,0 +1,63 @@
+
+# Running Apache Brooklin inside Karaf container
+
+The Apache Brooklyn Karaf based distribution lives in brooklyn-server/karaf/apache-brooklyn folder.
```
Worth adding a note on the status of the Karaf distro, with caveats about not everything necessarily working yet.

geomacy commented on guide/misc/osgi.md:
```diff
@@ -0,0 +1,63 @@
+
+You can use the standard ~/.brooklyn/brooklyn.properties file to configure Brooklyn. Alternatively
+create etc/brooklyn.cfg inside the distribution folder (same file format). The keys in the former override
+those in the latter.
```
I'd reword this to say 'use <install-dir>/etc/brooklyn.cfg' but making a note that ~/.brooklyn/brooklyn.properties is still supported for backward compatibility.   But also explain that values in brooklyn.cfg can be viewed and managed dynamically via the OSGI configuration admin commands in Karaf, e.g. config:property-set, so this is better suited for working in an OSGI environment.

geomacy commented on guide/misc/osgi.md:
```diff
@@ -0,0 +1,63 @@
+create etc/brooklyn.cfg inside the distribution folder (same file format). The keys in the former override
+those in the latter.
+
+Web console related configuration is done through the corresponding Karaf mechanisms:
```
Need a blank line after this one if the bullets below are to render properly.

geomacy commented:
What do we do about the numerous other pages where 'brooklyn.properties' is referenced?   Is it best to leave them as-is until Brooklyn moves to a Karaf-only distribution?

geomacy commented:
that's all the comments from me...

neykov commented:
Added a new commit addressing the comments so it's clear what the changes are. Will squash before merging.
@sjcorbett The instructions are mostly one-liners copy & pasted from my workflow, so no need to go through them. What I'd be really interested in is an "outsider" testing the distribution and reporting his experience.
